src/content/developers/docs/scaling/optimistic-rollups/index.md

```diff
@@ -1,46 +1,254 @@
+
+Optimistic rollups are considered “optimistic” because they assume off-chain transactions are valid and don't publish proofs of validity for transaction batches posted on-chain. This separates optimistic rollups from [zero-knowledge rollups](/developers/docs/scaling/zk-rollups) that publish cryptographic [proofs of validity](/glossary/#validity-proof) for off-chain transactions. 
+
+Optimistic rollups instead rely on a fraud-proving scheme to detect malicious transactions. After a rollup batch is submitted on Ethereum, there's a time window (called a challenge period) during which anyone can challenge the validity of a rollup transaction by computing a [fraud proof](/glossary/#fraud-proof). 
+
+If the fraud proof succeeds, the rollup protocol reverses the invalid block and punishes the malicious operator. If the rollup batch remains unchallenged after the challenge period elapses, it is deemed valid and accepted on Ethereum. The rollup will advance to a new state and the operator can start adding new blocks. 
+
+## How do optimistic rollups interact with Ethereum? {#optimistic-rollups-and-Ethereum} 
+
+An optimistic rollup is an off-chain protocol managed by a set of smart contracts deployed on the Ethereum network. Transactions executed off-chain are added to a record stored in the main rollup contract. Like the Ethereum blockchain, this transaction record is immutable and forms the "optimistic rollup chain."
+
+Although optimistic rollups exist as separate protocols, they are secured by the Ethereum network. Among other things, Ethereum guarantees the correctness of a rollup’s off-chain computation and the availability of data behind the computation. 
+
+### Data availability {data-availability}
+
+As mentioned, optimistic rollups post transaction data to Ethereum as `calldata`. Since the rollup chain's execution is based on submitted transactions, anyone can use this information—anchored on Ethereum’s base layer—to execute the rollup’s state and verify the correctness of state transitions. . 
+
+Data availability is critical because without access to state data, challengers cannot construct fraud proofs to dispute invalid rollup operations. With Ethereum providing data availability, the risk of rollup operators getting away with malicious acts (e.g., submitting invalid blocks) reduces. 
+
+### Censorship resistance {#censorship-resistance}
+
+Optimistic rollups also rely on Ethereum for censorship resistance. In an optimistic rollup a centralized entity (the operator) is responsible for processing transactions and submitting rollup blocks to Ethereum. This has some implications: 
+
+- Rollup operators can censor users by going offline, refusing to produce blocks, or submit user transactions to L1. 
```

Comment:
How about "Rollup operators can censor users by going offline completely, or by refusing to either produce blocks that include certain transactions in them"?

Reply:
Yes I bought some eth last year to invest on MetaMask

Reply:
How do I recover my eth

Reply:
I think they rewrote the code to 256